azurerm/internal/services/redisenterprise/redis_enterprise_cluster_resource_test.go

```diff
@@ -0,0 +1,139 @@
+package redisenterprise_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance/check"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/redisenterprise/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+type RedisEnterpriseClusterResource struct{}
+
+func TestAccRedisEnterpriseCluster_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_redis_enterprise_cluster", "test")
+	r := RedisEnterpriseClusterResource{}
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.basic(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccRedisEnterpriseCluster_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_redis_enterprise_cluster", "test")
+	r := RedisEnterpriseClusterResource{}
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.basic(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func TestAccRedisEnterpriseCluster_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_redis_enterprise_cluster", "test")
+	r := RedisEnterpriseClusterResource{}
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.complete(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
```

Comment:
could we add a update going from basic -> complete -> basic here?

Reply:
Nope, update is "not implemented" as of yet... I had added that test until I figured out they had not implemented the update function yet. 😭